dali/operators/imgcodec/image_decoder.h

```diff
@@ -0,0 +1,796 @@
+// Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// limitations under the License.
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/core/mm/memory.h"
+#include "dali/operators/decoder/cache/cached_decoder_impl.h"
+#include "dali/operators/generic/slice/slice_attr.h"
+#include "dali/operators/image/crop/crop_attr.h"
+#include "dali/operators/image/crop/random_crop_attr.h"
+#include "dali/operators/imgcodec/util/convert.h"
+#include "dali/operators/imgcodec/util/convert_gpu.h"
+#include "dali/operators/imgcodec/util/convert_utils.h"
+#include "dali/operators/imgcodec/util/nvimagecodec_types.h"
+#include "dali/operators/imgcodec/util/output_shape.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/checkpointing/stateless_operator.h"
+#include "dali/pipeline/operator/operator.h"
+
+#ifndef DALI_OPERATORS_IMGCODEC_IMAGE_DECODER_H_
+#define DALI_OPERATORS_IMGCODEC_IMAGE_DECODER_H_
+
+namespace dali {
+namespace imgcodec {
+
+template <typename Backend>
+struct OutBackend {
+  using type = GPUBackend;
+};
+
+template <>
+struct OutBackend<CPUBackend> {
+  using type = CPUBackend;
+};
+
+
+static uint32_t verbosity2severity(int verbose) {
```

Comment:
nitpick: I guess `verbosity_to_severity` would be more in line with the styleguide